bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubBridgeHandler.java

```diff
@@ -0,0 +1,174 @@
+
+    public MetOfficeDataHubBridgeHandler(final Bridge bridge, IHttpClientProvider httpClientProvider,
+            @Reference TranslationProvider translationProvider, @Reference LocaleProvider localeProvider) {
+        super(bridge);
+        this.translationProvider = translationProvider;
+        this.localeProvider = localeProvider;
+        this.bundle = FrameworkUtil.getBundle(getClass());
+        this.httpClient = httpClientProvider.getHttpClient();
+    }
+
+    protected String getApiKey() {
+        return config.siteApiKey;
+    }
+
+    public String getLocalizedText(String key, @Nullable Object @Nullable... arguments) {
+        String result = translationProvider.getText(bundle, key, key, localeProvider.getLocale(), arguments);
+        return Objects.nonNull(result) ? result : key;
+    }
+
+    public void updateLimiterStats() {
+        final Map<String, String> newProps = new HashMap<>();
+        newProps.put(BRIDGE_PROP_FORECAST_REQUEST_COUNT, String.valueOf(forecastDataLimiter.getCurrentRequestCount()));
+        this.updateProperties(newProps);
+    }
+
+    private static long getMillisUntilMidnight() {
+        return Duration.between(LocalDateTime.now(), LocalDate.now().plusDays(1).atStartOfDay()).toMillis();
+    }
+
+    protected static long getMillisSinceDayStart() {
+        return Duration.between(LocalDate.now().atStartOfDay(), LocalDateTime.now()).toMillis();
+    }
+
+    @Override
+    public void initialize() {
+        updateLimiterStats();
+        config = getConfigAs(MetOfficeDataHubBridgeConfiguration.class);
+        forecastDataLimiter.updateLimit(config.siteRateDailyLimit);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduleResetDailyLimiters();
+
+        initTask = scheduler.schedule(() -> {
+            final PointType siteApiTestLocation = new PointType("51.5072,0.1276");
+            updateLimiterStats();
+
+            final Response.CompleteListener siteResponseListener = new BufferingResponseListener() { // 4.5kb buffer
+                @Override
+                public void onComplete(@Nullable Result result) {
+                    if (result != null && !result.isFailed()) {
+                        updateStatus(ThingStatus.ONLINE);
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                getLocalizedText("bridge.error.site-specific.auth-issue"));
+                    }
+                }
+            };
+            MetOfficeDataHubSiteApiHandler.sendAsyncSiteApiRequest(httpClient, true, getApiKey(), siteApiTestLocation,
+                    siteResponseListener);
+            initTask = null;
+        }, 1, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        cancelResetDailyLimiters();
+        if (initTask != null) {
+            initTask.cancel(true);
+            initTask = null;
```

Comment:
```suggestion
        ScheduledFuture<?> initTask = this.initTask;
        if (initTask != null) {
            initTask.cancel(true);
            this.initTask = null;
```

Reply:
Done in part of refactor thank you :) So should be good - if its not this one I saw another one and changed that.